compiler/compiler-core.cpp

```diff
@@ -494,6 +494,14 @@ const Index &CompilerCore::get_index() {
 }
 
 File *CompilerCore::get_file_info(std::string &&file_name) {
+  File *f = cpp_index.get_file(file_name);
+  if (f != nullptr) {
+    return f;
+  }
+//  printf("%s not found in index, creating\n", file_name.c_str());
+
+  static std::mutex mutex;
```

Comment:
nit: from my point of view it would be much better if this mutex was placed as CompilerCore field.

Reply:
Outdated, this function was refactored